tests/nrf5340_audio/sd_card/src/main.c

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+#include <zephyr/device.h>
+#include <zephyr/storage/disk_access.h>
+#include <modules/sd_card.h>
+#include <zephyr/fs/fs.h>
+#include <ff.h>
+#include <math.h>
+
+#define MKFS_DEV_ID "SD:"
+
+static int test_create_file(const char *path)
+{
+	int ret;
+	struct fs_file_t file;
+
+	fs_file_t_init(&file);
+	ret = fs_open(&file, path, FS_O_CREATE | FS_O_WRITE);
+	if (ret == 0) {
+		fs_write(&file, "Testfile", 8);
+		fs_close(&file);
+	} else {
+		TC_PRINT("Failed to open file\n");
+		return ret;
+	}
+
+	return 0;
+}
+
+static int dir_name_get(char *name, size_t name_size, uint32_t curr_level, uint32_t curr_dir_id,
+			uint8_t subdirs)
+{
+	int ret;
+
+	name[0] = '\0';
+	strcat(name, "/SD:");
+
+	struct dir_tree {
+		uint32_t peer_id;
+		uint32_t name;
+	};
+
+	struct dir_tree f_tree[5];
+
+	uint32_t on_level = curr_level;
+
+	f_tree[on_level].peer_id = curr_dir_id;
+	f_tree[on_level].name = curr_dir_id % subdirs;
+
+	while (on_level > 0) {
+		f_tree[on_level - 1].peer_id = f_tree[on_level].peer_id / subdirs;
+		f_tree[on_level - 1].name = f_tree[on_level - 1].peer_id % subdirs;
+		on_level--;
+	}
+
+	for (int i = 0; i < curr_level + 1; i++) {
+		char tmp[100] = {'\0'};
+
+		ret = sprintf(tmp, "/dir_%d", f_tree[i].name);
+		strcat(name, tmp);
+	}
+
+	return 0;
+}
+
+static bool directory_exists(const char *path)
+{
+	struct fs_dirent entry;
+	int ret = fs_stat(path, &entry);
+
+	return (ret == 0 && entry.type == FS_DIR_ENTRY_DIR);
+}
+
+static int test_disk_populate(uint8_t num_levels, uint8_t files_per_dir, uint8_t subdirs,
+			      char const *const extension)
+{
+	TC_PRINT("Writing files. Num levels %u, files_per_dir %u, dirs_pr_level %u\n", num_levels,
+		 files_per_dir, subdirs);
+
+	int ret;
+	int files_added = 0;
+	char current_path[200] = {'\0'};
+
+	for (int level = 0; level < num_levels; level++) {
+		uint32_t dirs_on_level = pow(subdirs, (level + 1));
+
+		for (int i = 0; i < dirs_on_level; i++) {
+			ret = dir_name_get(current_path, 200, level, i, subdirs);
+			zassert_equal(0, ret, "dir_name_get should return 0, %u", ret);
+			if (!directory_exists(current_path)) {
+				ret = fs_mkdir(current_path);
+				zassert_equal(0, ret, "fs_mkdir should return 0, %u", ret);
+			}
+
+			for (int j = 0; j < files_per_dir; j++) {
+				char filepath[1000] = {'\0'};
+				char filename[100] = {'\0'};
+
+				strcat(filepath, current_path);
+				sprintf(filename, "/file_%d%s", j, extension);
+				strcat(filepath, filename);
+				ret = test_create_file(filepath);
+				zassert_equal(0, ret, "test_create_file should return 0, %u", ret);
+				files_added++;
+			}
+		}
+	}
+
+	TC_PRINT("--------Done writing %d files--------\n", files_added);
+	return files_added;
+}
+
+static void *setup_fn(void)
```

Comment:
Does this function need to return a pointer? Since it will only ever return NULL.

Reply:
Yes, that is given by the test framework.